fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/runners/throttled/ThrottledIteratorTest.java

```diff
@@ -619,11 +638,14 @@ void testIteratorClosesIncompleteFutures() throws Exception {
         try (FDBRecordContext context = openContext()) {
             openSimpleRecordStore(context);
             storeBuilder = recordStore.asBuilder();
+            commit(context);
         }
         final ThrottledRetryingIterator.Builder<Integer> builder = ThrottledRetryingIterator.builder(fdb, cursorFactory, itemHandler)
                 .withTransactionInitNotification(initNotification);
         try (ThrottledRetryingIterator<Integer> iterator = builder.build()) {
             iterateAll = iterator.iterateAll(storeBuilder);
+            // "sleep" for 10 ms to allow futures to get scheduled
+            MoreAsyncUtil.delayedFuture(10, TimeUnit.MILLISECONDS).join();
```

Comment:
Can this test remove the sleep by doing something similar to the above, where the non-completing future awaits on a semaphore that isn't released until after the iterator is closed. Maybe you need two semaphores.